packages/cursorless-vscode/src/extension.ts

```diff
@@ -48,6 +49,29 @@ import {
 import { StatusBarItem } from "./StatusBarItem";
 import { vscodeApi } from "./vscodeApi";
 import { mkdir } from "fs/promises";
+import { CommandRunner, TestCaseRecorder } from "@cursorless/cursorless-engine";
+import { ReadOnlyHatMap } from "@cursorless/common";
+
+class TestCaseRecorderCommandRunnerDecorator implements CommandRunnerDecorator {
+  testCaseRecorder: TestCaseRecorder;
+
+  constructor(testCaseRecorder: TestCaseRecorder) {
+    this.testCaseRecorder = testCaseRecorder;
+  }
+  wrapCommandRunner(
+    readableHatMap: ReadOnlyHatMap,
+    commandRunner: CommandRunner,
+  ) {
+    if (this.testCaseRecorder.isActive()) {
```

Comment:
I believe if you move this if statement into the wrap function in the test case recorder itself, you could do away with this whole glue class.